node.go

```diff
@@ -602,3 +602,159 @@ func sortFields(fields []Field) {
 func compareFields(a, b Field) int {
 	return strings.Compare(a.Name(), b.Name())
 }
+
+func RewriteSchema(schema *Schema, replace func(path []string, node Node) Node) *Schema {
+	return NewSchema(schema.name, rewriteSchema(nil, schema.root, replace))
+}
+
+func rewriteSchema(path []string, node Node, replace func(path []string, node Node) Node) Node {
+	if node.Leaf() {
+		return replace(path, node)
+	}
+
+	// Types requiring special handling.
+	switch n := node.(type) {
+	case *structNode:
```

Comment:
structNode/Field - This must be recreated in order to maintain order of leafs to match struct definition (otherwise everything is reordered by name), and the index for field lookups.  structField has a custom `Value()` implementation that groupField does not satisfy.